application/src/main/java/org/opentripplanner/osm/model/OsmLevelFactory.java

```diff
@@ -0,0 +1,53 @@
+package org.opentripplanner.osm.model;
+
+import javax.annotation.Nullable;
+import org.opentripplanner.graph_builder.issue.api.DataImportIssueStore;
+import org.opentripplanner.osm.issues.FloorNumberUnknownAssumedGroundLevel;
+
+public class OsmLevelFactory {
+
+  public static final OsmLevel DEFAULT = new OsmLevel(0.0, "default level");
+
+  private final DataImportIssueStore issueStore;
+
+  public OsmLevelFactory(DataImportIssueStore issueStore) {
+    this.issueStore = issueStore;
+  }
+
+  /**
+   * Create an OsmLevel for an entity by parsing the 'level' and 'layer' tags.
+   * If the level is parsed from the 'level' tag, the 'level:ref' tag is used for a name.
```

Comment:
I keep forgetting the difference between level and layer. Can you please write a few words.

Alternatively, add `getLayer` and `getLevel` methods to `OsmEntity` and put the docs there.